src/fmu/settings/_resources/userlog_manager.py

```diff
@@ -0,0 +1,39 @@
+"""Manages an .fmu userlog file."""
+
+from __future__ import annotations
+
+import contextlib
+from pathlib import Path
+from typing import TYPE_CHECKING, Self
+
+from fmu.settings._resources.log_manager import LogManager
+from fmu.settings.models.log import Log, LogFileName
+from fmu.settings.models.user_info import UserInfo
+
+if TYPE_CHECKING:
+    # Avoid circular dependency for type hint in __init__ only
+    from fmu.settings._fmu_dir import (
+        FMUDirectoryBase,
+    )
+
+
+class UserlogManager(LogManager[UserInfo]):
+    """Manages the .fmu userlog file."""
+
+    def __init__(self: Self, fmu_dir: FMUDirectoryBase) -> None:
+        """Initializes the User log resource manager."""
+        super().__init__(fmu_dir, Log[UserInfo])
+
+        # If userlog.json exists from previous session, cache it and then delete it
+        # We want a fresh log each time
```

Comment:
I think this is reasonable. I also like how simple and expressive this is to do internally!